papers/bradley_dice/55_bradley_dice.rst

```diff
@@ -408,86 +422,100 @@ For example, a **signac** data space that describes weather data for multiple ci
+
 
 Synced Collections: Backend-agnostic, persistent, mutable data structures
 -------------------------------------------------------------------------
 
 Motivation
 ~~~~~~~~~~
 
-All of **signac**'s principal functions are designed around efficiently indexing a collection of directories.
-By organizing job directories by the hash of their state point, **signac** can perform many operations in constant time.
-To present a Pythonic API, state points are exposed via a dictionary-like interface, making it very easy to modify a state point and have that change transparently reflected in both the JSON file and the name of the corresponding directory (which is the state point's hash).
+At its core, **signac** is a tool for organizing and working with data on the filesystem, presenting a Pythonic interface for tasks like creating directories and modifying files.
+In particular, **signac** makes modifying the JSON files used to store a job's state points and documents as easy as working with Python dictionaries.
+Despite heavy optimization, when seeking to scale **signac** to ever-larger data spaces, we quickly realized that the most significant performance barrier was the overhead of parsing and modifying large numbers of text files.
+Unfortunately, the usage of JSON files in this manner was deeply embedded in our data model, making switching to a more performant backend without breaking APIs or severely complicating our data model a daunting task.
 
-The need to parse these JSON files for indexing and the complexity of modifying them represent the most significant barriers to scaling **signac**.
-Even in the absence of file modification, simply reading a large number of files to produce a database index becomes prohibitively expensive for large data spaces.
-Although various optimizations have incrementally improved **signac**'s scalability, an alternative means of storing the state point and associated metadata that circumvents the heavy I/O costs of our current approach has the potential to make a much larger impact.
-However, replacing individual JSON files as the primary data source for **signac** without breaking **signac**'s API would require a generic means for providing the same interface to the underlying index and metadata irrespective of the underlying storage mechanism.
-Once developed, however, such an API would abstract out enough of the internals of **signac** to enable other generalizations as well, such as making it relatively easy to support alternate (and nearly arbitrary) data space layouts.
+While attempting to overcome this hurdle, we identified a core problem that our current API for modifying JSON files was solving in a limited manner: namely, the creation of a dictionary-like interface to an underlying resource.
+Numerous other well-known Python packages (h5py and Zarr immediately come to mind) also use dictionary-like interfaces to make working with complex resources feel natural to Python users.
```

Comment:
```suggestion
Numerous other well-known Python packages (_e. g._ h5py and Zarr) also use dictionary-like interfaces to make working with complex resources feel natural to Python users.
```

Reply:
I am fine with the suggestion, but I don't think that e.g. is supposed to be italicized even if it is from a Latin phrase.
```suggestion
Numerous other well-known Python packages (e.g. h5py and Zarr) also use dictionary-like interfaces to make working with complex resources feel natural to Python users.
```

Reply:
Adapted with citations in 8d27b167.